active/ItemsRepeater/ItemsRepeater-FirstVisibleIndex.md

```diff
@@ -0,0 +1,124 @@
+
+# Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+FirstVisibleIndex, LastVisibleIndex,and RealizedItems are properties of ItemsRepeater that give you access to which indices of items within an ItemsRepeater are currently visible on the screen.
+
+The `FirstVisibleIndex` property returns the lowest index of the currently visible indices, which is the first visible index. The `LastVisibleIndex` property returns the highest index of the currently visible indices, which is the last visible index. For contiguous scenarios/layouts (uniform layouts), all items with indices between the first and last visible index are currently visible on the screen. For non-contiguous/non-uniform/nested layouts, the first and last visible index may not correlate to the first and last visible items on the screen.
+
+
+The `RealizedItems` property returns all items within the ItemsRepeater that are currently realized and visible on the screen, including items that may be cut off or not 100% visible. This property is helpful for determining which items are visible in non-contiguous and non-uniform layout scenarios.
+
+# Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+```csharp
+// Check if an item is currently visible in a uniform layout.
+int item_index = repeater.GetIndexFromElement(item);
+if (item_index > repeater.FirstVisibleIndex && item_index < repeater.LastVisibleIndex){
+    //item is visible, perform necessary actions
+}
+
+// In a non-contiguous layout, find the index of the first visible item.
+UIElement last_item = repeater.RealizedItems[0] as UIElement;
+int first_visible_idx = repeater.GetIndexFromElement(last_item);
+// first_visible_idx represents the first visible index, perform necessary actions on it. 
```

Comment:
Since `RealizedItems` is unsorted, `RealizedItems` is not guarenteed to be the first realized item, i.e. it has the lowest index out of all the realized items. Wouldn't be FirstVisibleIndex the appropiate way to get the first realized item?

Reply:
Thanks, I realized this after the discussion on sorting RealizedItems :) I'll fix this.